document/value.go

```diff
@@ -93,12 +93,12 @@ func (t ValueType) IsNumber() bool {
 
 // IsInteger returns true if t is a signed or unsigned integer of any size.
 func (t ValueType) IsInteger() bool {
-	return t >= Int8Value && t <= Int64Value || t == DurationValue
+	return t == IntegerValue || t == DurationValue
 }
 
-// IsFloat returns true if t is either a Float32 or Float64.
+// IsFloat returns true if t is a Double.
 func (t ValueType) IsFloat() bool {
```

Comment:
What about renaming this to `IsDouble`?

Reply:
This is a good question that I was asking myself for a while. But it depends on what we want to answer here:
- Do we want to know if the type is from the `float` family (historically we had Float32 and Float64 support)
- or do we want to know if the type is a double?

The first question doesn't really need to be answered since we only have one float type now. 
The second question can be answered by doing `v.Type == document.DoubleValue`.

I think this function will get removed eventually along others of this package, but I think it deserves another PR

Reply:
> I think it deserves another PR

Definitely. Let's keep it as it is for now if it will be removed later on.